sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -352,8 +352,9 @@ case class AdaptiveSparkPlanExec(
           val newCost = costEvaluator.evaluateCost(newPhysicalPlan)
           if (newCost < origCost ||
             (newCost == origCost && currentPhysicalPlan != newPhysicalPlan)) {
-            logOnLevel("Plan changed:\n" +
-              sideBySide(currentPhysicalPlan.treeString, newPhysicalPlan.treeString).mkString("\n"))
```

Comment:
hmmm .. I think we should actually make `Logging.logTrace` call by value so the inside ones are not evaluated

Reply:
Let me take a quick look. Feel free to merge this first if you prefer. Seems fine